index.js

```diff
@@ -1091,6 +1110,10 @@ Connection.prototype._apiRequest = function(opts, callback) {
         e = new Error(body);
         e.errorCode = body;
         e.body = body;
+      } else if (_.isObject(body) && findValueForKeyInObject(body, 'message', 0, 5) && findValueForKeyInObject(body, 'errorCode', 0, 5)) {
```

Comment:
In the case that the body is an object, use the above recursive function to see if both a `message` and an `errorCode` exist. If they do, use those to create the new `Error` object. This will prevent consumers from getting an appropriate status code with a relatively useless `"Salesforce returned an unrecognized error 409"` for example.